app/models/system.rb

```diff
@@ -20,7 +20,16 @@ class System < ApplicationRecord
   alias_attribute :scc_synced_at, :scc_registered_at
 
   def init
-    self.login ||= System.generate_secure_login
+    self.login ||= if system_token.presence
+                     # if we are in the public cloud
+                     # system_token should be not nil
+                     # obfuscate the instance id
+                     Base64.strict_encode64(system_token)
```

Comment:
I do not really see a reason to encode the instance ID

Reply:
It is the same, guessing another instances instance id is going to be about as successful as guessing someone else' generate user name. In any event the user name alone is useless and the underlying system will still fail access when it gets re-verified when the cache expires. I cannot think of a way to exploit the knowledge that the user name is the instance id.